### PR TITLE
Show "all versions" when a block has min = 0 and max = *

### LINKED ISSUE #9250: Show "all versions" when a block has min = 0 and max = *
See: https://github.com/mozilla/addons-frontend/pull/9236#discussion_r394357051

---

Instead of showing:

> Versions blocked: 0 to *

let's show:

> Versions blocked: all versions

in the newly added "block" page.

URL of a block in local dev (when using `yarn amo:dev` locally): http://localhost:3000/en-US/firefox/blocked/%7B46551EC9-40F0-4e47-8E18-8E5CF550CFB8%7D/

### PR DESCRIPTION
**Pull request by muskangupta-iitr**
Fixes #9250 

Changed the text of versions when `min_version` is `0` and `max_version` is `*`.

### CODE REVIEW

**Comment by muskangupta-iitr** [comment]
@willdurand Please review!

File: src/amo/pages/Block/index.js [hunk 1]
```diff
@@ -178,12 +178,16 @@ export class BlockBase extends React.Component<InternalProps> {
...
+                block.min_version === '0' && block.max_version === '*' ? (
+                  i18n.sprintf(
+                    i18n.gettext('Versions blocked: %(min)s to %(max)s.'),
+                    {
+                      min: block.min_version,
+                      max: block.max_version,
+                    },
+                  )
+                ) : (
+                  i18n.gettext('Versions blocked: all versions.')
```
Comment (willdurand) [inline src/amo/pages/Block/index.js]: Shouldn't it be the other way around?

Write a test case in `TestBlock.js` to make sure it works as intended.

Comment (muskangupta-iitr) [reply src/amo/pages/Block/index.js]: Oh sorry! Yes I will update the pr.

Comment (muskangupta-iitr) [reply src/amo/pages/Block/index.js]: Yes I will write the test.

**Comment by muskangupta-iitr** [comment]
@willdurand Done :+1:

File: src/amo/pages/Block/index.js [hunk 2]
```diff
@@ -102,6 +102,23 @@ export class BlockBase extends React.Component<InternalProps> {
     return content;
   }
 
+  renderBlockData() {
```
Comment (willdurand) [inline src/amo/pages/Block/index.js]: We can call this method: `renderVersions`

File: tests/unit/amo/pages/TestBlock.js [hunk 3]
```diff
@@ -213,7 +213,22 @@ describe(__filename, () => {
     expect(root.find('.Block-reason')).toHaveLength(0);
   });
 
-  it('renders the min/max versions', () => {
+  it('renders the min/max versions if min is 0 and max is *', () => {
```
Comment (willdurand) [inline tests/unit/amo/pages/TestBlock.js]: it `renders "all versions" when min is 0 and max is *`

File: src/amo/pages/Block/index.js (hunk 2 above)
Comment (muskangupta-iitr) [reply src/amo/pages/Block/index.js]: Ohk!

File: tests/unit/amo/pages/TestBlock.js (hunk 3 above)
Comment (muskangupta-iitr) [reply tests/unit/amo/pages/TestBlock.js]: :+1:

**Comment by muskangupta-iitr** [comment]
@willdurand Done!

File: src/amo/pages/Block/index.js [hunk 4]
```diff
@@ -102,6 +102,23 @@ export class BlockBase extends React.Component<InternalProps> {
     return content;
   }
 
+  renderVersion() {
```
Comment (willdurand) [inline src/amo/pages/Block/index.js]: Can you use the plural form here? (because we render the min and max versions)

Comment (muskangupta-iitr) [reply src/amo/pages/Block/index.js]: No Problem!

**Comment by muskangupta-iitr** [comment]
@willdurand Done!

**Comment by willdurand** [comment]
Good job @muskangupta-iitr, thanks!

**Pull request merged**